### PR TITLE
persist data without server

### PR DESCRIPTION
**Pull request by Dillon-Roller**
## Summary by CodeRabbit

- **New Features**
	- Introduced theme configuration management, allowing users to save and load their preferred themes.
- **Refactor**
	- Updated the `Settings` and `Widget` components to support the new theme configuration management.
- **Chores**
	- Added the `confy` crate for configuration management and serialization/deserialization of configuration data.

### CODE REVIEW

File: client/src/components/settings.rs [hunk 1]
```diff
@@ -41,6 +59,40 @@ pub(crate) fn Settings(
...
+}
+
+pub fn load_theme_from_config(theme_type: ThemeType) -> String {
+    let cfg: ThemeConfig = confy::load(APP_NAME, CONFIG_NAME).unwrap_or_default();
+
+    match theme_type {
+        ThemeType::Board => cfg.board_theme,
+        ThemeType::Piece => cfg.piece_theme,
+    }
+}
```
[the remark this answers is not in the document]
Comment (Dillon-Roller) [reply client/src/components/settings.rs:84]: unwrap_or_default() returns the default if it panics.

**Comment by DanielAlbl** [comment]
@coderabbitai I tried compiling this for wasm and got this error in the bowser:
```client-b8c6854fb3c22478.js:446 panicked at client/src/components/settings.rs:94:46:
failed to store the theme configuration: BadConfigDirectory("could not determine home directory path")```

File: client/src/components/settings.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dillon-Roller) [reply client/src/components/settings.rs:84]: default does not panic

**Comment by DanielAlbl** [comment]
So if you want to put the themes into one settings hook in this PR, otherwise this is good.

**Review by DanielAlbl — APPROVED** [review]

**Pull request merged**